### PR TITLE
Fix similar issues tests and empty issue titles

### PR DESCRIPTION
**Pull request by sharkykh**
- Fix find_similar_issues logging and tests
- Fix empty issue title and add test
  Fixes titles like #4369

### CODE REVIEW

File: medusa/issue_submitter.py [hunk 1]
```diff
@@ -121,19 +126,25 @@ def find_similar_issues(cls, github_repo, loglines, max_age=timedelta(days=180))
...
-                log.debug('Searching for issues similar to: {0}', log_title)
 
                 # Apply diff ratio overrides on first-matched basis, default = 0.9
                 diff_ratio = next((override[1] for override in cls.TITLE_DIFF_RATIO_OVERRIDES
                                    if override[0] in log_title.lower()), 0.9)
 
                 if cls.similar(log_title, issue_title, diff_ratio):
+                    log.debug(
+                        'Found issue #{number} ({issue})'
+                        ' to be similar ({ratio:.0%}}) to {log}',
```
Comment (medariox) [inline medusa/issue_submitter.py]: `{ratio:.0%}}` one `}` too much.

File: medusa/issue_submitter.py [hunk 2]
```diff
@@ -183,6 +194,7 @@ def result(message, level=logging.WARNING):
         except RateLimitExceededException:
             return result(self.RATE_LIMIT)
         except (GithubException, IOError) as error:
+            log.debug('Issue submitter failed with error: {!r}', error)
```
Comment (medariox) [inline medusa/issue_submitter.py]: I know that it is optional, but I'd still go for `{0!r}`.

File: medusa/issue_submitter.py (hunk 1 above)
Comment (sharkykh) [reply medusa/issue_submitter.py]: Oops, good catch! Thanks!

File: medusa/logger/__init__.py [hunk 3]
```diff
@@ -346,7 +346,10 @@ def key(self):
     @property
     def issue_title(self):
         """Return the expected issue title for this logline."""
-        result = self.traceback_lines[-1] if self.traceback_lines else self.message
+        if self.traceback_lines:
+            result = next((line for line in reversed(self.traceback_lines) if line.strip()), self.message)
```
Comment (medariox) [inline medusa/logger/__init__.py]: Is the `if` statement really needed?

Comment (sharkykh) [reply medusa/logger/__init__.py]: Yes. Traceback lines can have an empty line, causing issue titles like this:
https://github.com/pymedusa/Medusa/issues/4369

**Pull request merged**